### PR TITLE
[Enhancement] Adding unit test for state-mixin

### LINKED ISSUE #2: tests

### PR DESCRIPTION
**Pull request by thoov**
Adding unit test for the state mixin. Fixes #2.

I will be sending another PR later with acceptance test covering the demo app + an upgraded demo example. 

First pass was done on #11

### CODE REVIEW

**Comment by thoov** [comment]
@stefanpenner review

File: tests/unit/state-mixin-test.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  subject.set('model', 'person1');
+  subject.set('state.firstName', 'John');
+
+  subject.set('model', 'person2');
+  subject.set('state.firstName', 'Jane');
+
+  subject.set('model', 'person1');
+  subject.get('state');
+
+  // teardownState calls destory which is scheduled
```
Comment (jayphelps) [inline tests/unit/state-mixin-test.js]: s/destory/destroy

Comment (thoov) [inline tests/unit/state-mixin-test.js]: :+1:

**Comment by jayphelps** [comment]
LGTM

**Comment by stefanpenner** [comment]
LGTM

**Pull request merged**